frontend/src/components/homepage/Filter.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+
+import ToggleButton from "../navigation/ToggleButton";
+import "./Filter.css";
+
+export type FilterProps = {
+  approvedLanguages: { [name: string]: number };
+  level: number;
+  setLevel: (newState: number) => void;
+  language: string;
+  setLanguage: (newState: string) => void;
+  role: boolean;
+  setRole: (newState: boolean) => void;
+};
+
+const Filter = ({
+  approvedLanguages,
+  level,
+  setLevel,
+  language,
+  setLanguage,
+  role,
+  setRole,
+}: FilterProps) => {
+  // language state in Homepage can't default to user context value
+  // set the default language from user approvedLanguages here
+  if (!Object.keys(approvedLanguages).includes(language)) {
```

Comment:
Does `!(language in approvedLanguages)` not work?

Reply:
(code is outdated now) but yee that works too!